Automattic/portkey-android
# Fix crash in added views

Closes #291 

Builds on top of PR #290

2 things were happening:

1. if an exception occurred in saving an Image-based Story frame and this made removing their AddedViews from the ghost (offscreen) photoEditorView impossible (as it would derail execution to the Exception handler), then those AddedViews would be left depending on that ghost view, and once we want to add them back it would be impossible, producing the crash. For this, we added a `finally` block to the exception handler to make sure to release those AddedViews, in  601463b.

2. when the service finishes working, the EventBus listener in the ComposeLoopFrameActivity would try and re-select the current frame, re-adding the AddedViews to the real (visible) photoEditorView and as such, leaving them impossible to be re-added to a new instance of photoEdiitorView later (as it would be the case when you need to manage errors, opening a new instance of the Activity from an error notification or snackbar). This frame selection refresh behavior we have in that listener was an old behavior we wanted to have before we decided to force exiting the Activity once the Service has been triggered (as per our use case) so, removed that code altogether in 6d5df8c

To test:

0. add an artificial crash, for example  add a `throw Exception("this is a test") ` right before [this line](https://github.com/Automattic/portkey-android/pull/292/files#diff-4053f04ce9e16619ae1679b34cb98dbfR119), this way the error handling mechanism takes course. 
1. verify the steps described in #291 don't lead to a crash anymore.
